### PR TITLE
Feature distance sensor interface

### PR DESCRIPTION
**Pull request by Gian1080**
Let us move this quickly, so leads have some time to perhaps merge it with master.

### CODE REVIEW

File: code/src/lidar_distance_sensor_c.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        frame.sensor_pov = 0;
+        frame.count = 45;
+
+        // 16 packets for one 360 degree measurement
+        for (int i = 0; i < 16; i++) {
+            lidar.receive_packet();
+        }
+        
+        // save a measurement every 8 degrees.
+        for (int i = 0; i < 45; i++) {
```
Comment (Official12345) [inline code/src/lidar_distance_sensor_c.cpp:17]: I get where the 45 is for but now this is a magic number. can this be in a centralized place?

File: code/src/lidar_distance_sensor_c.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+namespace r2d2::distance
+{
+    void lidar_distance_sensor_c::fill_distance_frame(frame_distance_s& frame){
+
+        frame.sensor_type = 0; // 0 = lidar, 1 = sonar
+        frame.sensor_pov = 0;
+        frame.count = 45;
+
+        // 16 packets for one 360 degree measurement
+        for (int i = 0; i < 16; i++) {
```
Comment (Official12345) [inline code/src/lidar_distance_sensor_c.cpp:12]: I get where the 16 is for but now this is a magic number. can this be in a centralized place?

**Review by Official12345 — CHANGES_REQUESTED** [review]
See local comments, and the lidar_distance_sensor_c members could use comments

File: code/src/lidar_distance_sensor_c.cpp (hunk 1 above)
Comment (Gian1080) [reply code/src/lidar_distance_sensor_c.cpp:17]: The comment should be enough, might have been 360/8, but clear nonetheless

File: code/src/lidar_distance_sensor_c.cpp (hunk 2 above)
Comment (Gian1080) [reply code/src/lidar_distance_sensor_c.cpp:12]: that might make it more unclear, imo

File: code/src/lidar_distance_sensor_c.cpp (hunk 1 above)
Comment (Official12345) [reply code/src/lidar_distance_sensor_c.cpp:17]: it is bad practice to put hard coded numbers in loops

Comment (Gian1080) [reply code/src/lidar_distance_sensor_c.cpp:17]: It should have been dynamic int that changes depending on the resolution the user wanted, you are right, but due to a lack of time we can not properly implement it.

**Review by Official12345 — APPROVED** [review]

**Pull request merged**